crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -30,21 +35,49 @@ pub(crate) fn run(args: ArgMatches) -> Result<(), CliError> {
     let mut proof_dir = program_dir.clone();
     proof_dir.push(PROOFS_DIR);
 
-    prove_with_path(proof_name, program_dir, proof_dir, check_proof, show_ssa, allow_warnings)?;
+    let mut circuit_build_path = program_dir.clone();
+    circuit_build_path.push(TARGET_DIR);
+    circuit_build_path.push(circuit_name);
+
+    prove_with_path(
+        proof_name,
+        program_dir,
+        proof_dir,
+        circuit_build_path,
+        check_proof,
+        show_ssa,
+        allow_warnings,
+    )?;
 
     Ok(())
 }
 
+#[allow(deprecated)]
```

Comment:
Noted above, using the hash method from acvm should allow you to get rid of this clippy lint